include/wx/private/socket.h

```diff
@@ -322,6 +318,11 @@ class wxSocketImpl
     // get the associated socket flags
     wxSocketFlags GetSocketFlags() const { return m_wxsocket->GetFlags(); }
 
+    // get the error code corresponding to the last operation
+    virtual wxSocketError GetLastError() = 0;
```

Comment:
Sorry, the new function name/semantics don't make much sense to me. AFAICS it should become
```suggestion
    // update the error code to correspond to the outcome of the last operation
    virtual void UpdateLastError() = 0;
```

as we don't use its return value and it modifies (and doesn't just "get") the error.